torch_geometric_temporal/nn/convolutional/mtgnn.py

```diff
@@ -0,0 +1,445 @@
+        if xd is not None:
+            self._static_feature_dim = xd
+            self._linear1 = nn.Linear(xd, dim)
+            self._linear2 = nn.Linear(xd, dim)
+        else:
+            self._embedding1 = nn.Embedding(nnodes, dim)
+            self._embedding2 = nn.Embedding(nnodes, dim)
+            self._linear1 = nn.Linear(dim, dim)
+            self._linear2 = nn.Linear(dim, dim)
+
+        self._k = k
+        self._alpha = alpha
+
+    def forward(self, idx: torch.LongTensor, static_feat: Optional[torch.FloatTensor]=None) -> torch.FloatTensor:
+        """
+        Making a forward pass to construct an adjacency matrix from node embeddings.
+
+        Arg types:
+            * **idx** (Pytorch Long Tensor) - Input indices, a permutation of the number of nodes, default None (no permutation).
+            * **static_feat** (Pytorch Float Tensor, optional) - Static feature, default None.
+        Return types:
+            * **adj** (PyTorch Float Tensor) - Adjacency matrix constructed from node embeddings.
+        """
+
+        if static_feat is None:
+            nodevec1 = self._embedding1(idx)
+            nodevec2 = self._embedding2(idx)
+        else:
+            assert static_feat.shape[1] == self._static_feature_dim
+            nodevec1 = static_feat[idx, :]
+            nodevec2 = nodevec1
+
+        nodevec1 = torch.tanh(self._alpha*self._linear1(nodevec1))
+        nodevec2 = torch.tanh(self._alpha*self._linear2(nodevec2))
+
+        a = torch.mm(nodevec1, nodevec2.transpose(1, 0)) - \
+            torch.mm(nodevec2, nodevec1.transpose(1, 0))
+        adj = F.relu(torch.tanh(self._alpha*a))
+        mask = torch.zeros(idx.size(0), idx.size(0)).to(adj.device)
+        mask.fill_(float('0'))
+        s1, t1 = adj.topk(self._k, 1)
+        mask.scatter_(1, t1, s1.fill_(1))
+        adj = adj*mask
+        return adj
+
+
+class LayerNorm(nn.Module):
+    __constants__ = ['normalized_shape', 'weight',
+                     'bias', 'eps', 'elementwise_affine']
+    r"""An implementation of the layer normalization layer.
+    For details see this paper: `"Connecting the Dots: Multivariate Time Series Forecasting with Graph Neural Networks." 
+    <https://arxiv.org/pdf/2005.11650.pdf>`_
+
+    Args:
+        normalized_shape (int): Input shape from an expected input of size.
+        eps (float, optional): Value added to the denominator for numerical stability. Default: 1e-5.
+        elementwise_affine (bool, optional): Whether to conduct elementwise affine transformation or not. Default: True.
+    """
+    def __init__(self, normalized_shape: int, eps: float=1e-5, elementwise_affine: bool=True):
+        super(LayerNorm, self).__init__()
+        if isinstance(normalized_shape, numbers.Integral):
+            normalized_shape = (normalized_shape,)
+        self._normlizationalized_shape = tuple(normalized_shape)
+        self._eps = eps
+        self._elementwise_affine = elementwise_affine
```

Comment:
Can you move out these to a private method?

Reply:
Do you mean putting the whole LayerNormalization into somewhere?